ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/network/DefaultP2PNetwork.java

```diff
@@ -383,11 +399,14 @@ void checkMaintainedConnectionPeers() {
   @VisibleForTesting
   void attemptPeerConnections() {
     LOG.trace("Initiating connections to discovered peers.");
-    rlpxAgent.connect(
+    final Stream<DiscoveryPeer> toTry =
         streamDiscoveredPeers()
             .filter(peer -> peer.getStatus() == PeerDiscoveryStatus.BONDED)
+            .peek(peer -> numBondedPeers.getAndIncrement())
```

Comment:
why do increment the number of bonded peers every time we attempt to connect?

Reply:
removed